python/ray/rllib/models/pytorch/fcnet.py

```diff
@@ -3,30 +3,27 @@
 from __future__ import print_function
 
 import logging
-
-from ray.rllib.models.pytorch.model import Model, SlimFC
-from ray.rllib.models.pytorch.misc import normc_initializer
+import numpy as np
 import torch.nn as nn
 
-logger = logging.getLogger(__name__)
+from ray.rllib.models.pytorch.model import TorchModel
+from ray.rllib.models.pytorch.misc import normc_initializer, SlimFC, \
+    _get_activation_fn
+from ray.rllib.utils.annotations import override
 
+logger = logging.getLogger(__name__)
 
-class FullyConnectedNetwork(Model):
-    """TODO(rliaw): Logits, Value should both be contained here"""
 
-    def _build_layers(self, inputs, num_outputs, options):
-        assert type(inputs) is int
-        hiddens = options.get("fcnet_hiddens", [256, 256])
-        fcnet_activation = options.get("fcnet_activation", "tanh")
-        activation = None
-        if fcnet_activation == "tanh":
-            activation = nn.Tanh
-        elif fcnet_activation == "relu":
-            activation = nn.ReLU
-        logger.info("Constructing fcnet {} {}".format(hiddens, activation))
+class FullyConnectedNetwork(TorchModel):
+    """Generic fully connected network."""
 
+    def __init__(self, obs_space, num_outputs, options):
+        TorchModel.__init__(self, obs_space, num_outputs, options)
+        hiddens = options.get("fcnet_hiddens")
```

Comment:
should this have a default?

Reply:
right now, if not provided, this fails at line 26 (which you may as well make this `options["fcnet_hiddens"]`

Reply:
It's filled in by agent, so it should be ok either way.